diff --git a/internal/customtypes/localized_string.go b/internal/customtypes/localized_string.go
index 37dc32c..d4b5eb0 100644
--- a/internal/customtypes/localized_string.go
+++ b/internal/customtypes/localized_string.go
@@ -44,3 +44,3 @@ func LocalizedString(opts LocalizedStringOpts) schema.MapAttribute {
 		Optional:   opts.Optional,
-		CustomType: NewLocalizedStringType(),
+		ElementType: types.StringType,
 		Validators: []validator.Map{
diff --git a/internal/resources/attribute_group/resource.go b/internal/resources/attribute_group/resource.go
index 2002433..7f7cfc6 100644
--- a/internal/resources/attribute_group/resource.go
+++ b/internal/resources/attribute_group/resource.go
@@ -7,3 +7,2 @@ import (
 	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
-	"github.com/labd/terraform-provider-commercetools/internal/customtypes"
 	"regexp"
@@ -69,3 +68,3 @@ func (r *Resource) Schema(_ context.Context, _ resource.SchemaRequest, resp *res
 			"name": schema.MapAttribute{
-				CustomType:          customtypes.NewLocalizedStringType(),
+				ElementType: 		 types.StringType,
 				Description:         "Name of the State as localized string.",
@@ -75,3 +74,3 @@ func (r *Resource) Schema(_ context.Context, _ resource.SchemaRequest, resp *res
 			"description": schema.MapAttribute{
-				CustomType:          customtypes.NewLocalizedStringType(),
+				ElementType: 		 types.StringType,
 				Description:         "Description of the State as localized string.",
diff --git a/internal/resources/product_selection/resource.go b/internal/resources/product_selection/resource.go
index d3b467a..458267d 100644
--- a/internal/resources/product_selection/resource.go
+++ b/internal/resources/product_selection/resource.go
@@ -18,3 +18,3 @@ import (
 
-	"github.com/labd/terraform-provider-commercetools/internal/customtypes"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 	"github.com/labd/terraform-provider-commercetools/internal/utils"
@@ -73,3 +73,3 @@ func (*productSelectionResource) Schema(_ context.Context, req resource.SchemaRe
 			"name": schema.MapAttribute{
-				CustomType:          customtypes.NewLocalizedStringType(),
+				ElementType: 		 types.StringType,
 				Description:         "Name of the ProductSelection.",
diff --git a/internal/resources/state/resource.go b/internal/resources/state/resource.go
index dd28402..f4dfe90 100644
--- a/internal/resources/state/resource.go
+++ b/internal/resources/state/resource.go
@@ -20,3 +20,2 @@ import (
 
-	"github.com/labd/terraform-provider-commercetools/internal/customtypes"
 	"github.com/labd/terraform-provider-commercetools/internal/utils"
@@ -64,3 +63,3 @@ func (r *stateResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
 			"name": schema.MapAttribute{
-				CustomType:          customtypes.NewLocalizedStringType(),
+				ElementType: 		 types.StringType,
 				Description:         "Name of the State as localized string.",
@@ -70,3 +69,3 @@ func (r *stateResource) Schema(_ context.Context, _ resource.SchemaRequest, resp
 			"description": schema.MapAttribute{
-				CustomType:          customtypes.NewLocalizedStringType(),
+				ElementType: 		 types.StringType,
 				Description:         "Description of the State as localized string.",
